core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestCompressionConfiguration.java

```diff
@@ -0,0 +1,139 @@
+        return minimumCompressionThresholdInBytes;
+    }
+
+    /**
+     * Create a {@link RequestCompressionConfiguration.Builder}, used to create a {@link RequestCompressionConfiguration}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        RequestCompressionConfiguration that = (RequestCompressionConfiguration) o;
+
+        if (!requestCompressionEnabled.equals(that.requestCompressionEnabled)) {
+            return false;
+        }
+        return Objects.equals(minimumCompressionThresholdInBytes, that.minimumCompressionThresholdInBytes);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = requestCompressionEnabled != null ? requestCompressionEnabled.hashCode() : 0;
+        result = 31 * result + (minimumCompressionThresholdInBytes != null ? minimumCompressionThresholdInBytes.hashCode() : 0);
+        return result;
+    }
+
+
+    public interface Builder extends CopyableBuilder<Builder, RequestCompressionConfiguration> {
+
+        /**
+         * Configures whether request compression is enabled or not. The default value is true.
+         *
+         * @param requestCompressionEnabled
+         * @return This object for method chaining.
+         */
+        Builder requestCompressionEnabled(Boolean requestCompressionEnabled);
+
+        /**
+         * Configures the minimum compression threshold, inclusive, in bytes. The default value is 10_240. The value must be
+         * non-negative and no greater than 10_485_760.
+         *
+         * @param minimumCompressionThresholdInBytes
+         * @return This object for method chaining.
+         */
+        Builder minimumCompressionThresholdInBytes(Integer minimumCompressionThresholdInBytes);
+    }
+
+    private static final class DefaultBuilder implements Builder {
+        private Boolean requestCompressionEnabled;
+        private Integer minimumCompressionThresholdInBytes;
+
+        private DefaultBuilder() {
+        }
+
+        private DefaultBuilder(RequestCompressionConfiguration requestCompressionConfiguration) {
+            this.requestCompressionEnabled = requestCompressionConfiguration.requestCompressionEnabled;
+            this.minimumCompressionThresholdInBytes = requestCompressionConfiguration.minimumCompressionThresholdInBytes;
+        }
+
+        @Override
+        public Builder requestCompressionEnabled(Boolean requestCompressionEnabled) {
+            this.requestCompressionEnabled = requestCompressionEnabled;
+            return this;
+        }
+
+        @Override
+        public Builder minimumCompressionThresholdInBytes(Integer minimumCompressionThresholdInBytes) {
```

Comment:
Should this be Long to be future proof?

Reply:
discussed offline, specification states integer value as must